test/integration/protocol_integration_test.cc

```diff
@@ -2445,6 +2445,69 @@ TEST_P(DownstreamProtocolIntegrationTest, BasicMaxStreamTimeoutLegacy) {
   EXPECT_THAT(waitForAccessLog(access_log_name_), HasSubstr("max_duration_timeout"));
 }
 
+TEST_P(DownstreamProtocolIntegrationTest, MaxRequestsPerConnectionNotReached) {
```

Comment:
I think this case can be combined with the next one; just validate that the stat is zero after the first request.

Reply:
Done.